services/system/Transact/src/RTransact.cpp

```diff
@@ -184,122 +184,244 @@ void RTransact::onTrx(const Checksum256& id, psio::view<const TransactionTrace>
    check(getSender() == AccountNumber{}, "Wrong sender");
    printf("trace size: %zu\n", find_view_span(trace).size());
 
-   TransactionTraceRef pruned = PruneTrace{true}(trace);
+   auto    clients = Subjective{}.open<TraceClientTable>();
+   uint8_t waitFor = WaitFor::final_flag;
+   PSIBASE_SUBJECTIVE_TX
+   {
+      if (auto row = clients.get(id))
+      {
+         waitFor = row->waitFor;
+      }
+   }
+
+   if (waitFor == WaitFor::final_flag)
+   {
+      auto currentBlock = to<Transact>().currentBlock();
+      auto blockTxs     = Subjective{}.open<BlockTxsTable>();
+      PSIBASE_SUBJECTIVE_TX
+      {
+         blockTxs.put(BlockTxRecord{
+             .id       = id,
+             .blockNum = currentBlock.blockNum,
```

Comment:
blockNum is only meaningful for successful transactions.

Reply:
Currently, it's set but unused for failed transactions. Do you want me to make it an optional and only set it on successful? Or two separate tables?

Reply:
Separated